Implementing batch/parallel downloads

ola-rozenfeld opened the pull request:
Fixing #38, #41.

Making sure to not read too much into memory at the same time.

ramymedhat commented on go/client/cas_test.go:
```diff
@@ -774,6 +714,143 @@ func TestDownloadActionOutputs(t *testing.T) {
+		},
+		{
+			name:      "small batches of big blobs",
+			sizes:     []int{1*mb + 1, 1*mb + 1, 1*mb + 1, 1*mb + 1, 1*mb + 1, 1*mb + 1, 1*mb + 1},
```
Can you also use MaxBatchSz here?

ola-rozenfeld replied:
Done, PTAL.

ola-rozenfeld commented:
Wow, this is very interesting -- I was wondering why I don't have tests that my executable files are actually executable. So I added them. The tests pass via Bazel, but fail via go test :-/
I suspect I ran into this before, and couldn't figure it out, that's why I removed the lines. Do you remember anything about that?
